Source/Core/Core/BootManager.cpp

```diff
@@ -378,9 +378,9 @@ bool BootCore(const std::string& _rFilename)
   // Apply overrides
   // Some NTSC GameCube games such as Baten Kaitos react strangely to
   // language settings that would be invalid on an NTSC system
-  if (!StartUp.bOverrideGCLanguage && ntsc)
+  if (!StartUp.bWii && !StartUp.bOverrideGCLanguage && !StartUp.GCLanguageAndRegionCompatible())
```

Comment:
Going by the setting only appearing on the GameCube config panel, the option itself only being called "OverrideGCLanguage", and the comment directly above this only referring to GameCube games it seems like this feature was specifically intended for only the GameCube.